### PR TITLE
Some more deprecated function calls are replaced

### PR DESCRIPTION
**Pull request by annando**
Several more function calls are now replaced and some legacy functions are completely removed now.

### CODE REVIEW

File: src/Module/Contact/Profile.php [hunk 1]
```diff
@@ -24,6 +24,7 @@
 use Friendica\App;
 use Friendica\BaseModule;
 use Friendica\Contact\LocalRelationship;
+use Friendica\Contact\LocalRelationship\Entity\LocalRelationship as EntityLocalRelationship;
```
Comment (MrPetovan) [inline src/Module/Contact/Profile.php]: Please rename the alias to `LocalRelationshipEntity` to satisfy the English order of qualifier.
```suggestion
use Friendica\Contact\LocalRelationship\Entity\LocalRelationship as LocalRelationshipEntity;
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Good job, it looks like `BaseModule` could use a `UserSession` dependency, but it will take a fair bit of work to get it in given the inheritance graph.

**Comment by annando** [comment]
Yeah, there will be some work to get rid of `DI` in these classes.

**Review by MrPetovan — APPROVED** [review]

File: src/App.php [hunk 2]
```diff
@@ -134,42 +134,6 @@ class App
...
-	public function getLoggedInUserId(): int
-	{
-		return $this->session->getLocalUserId();
-	}
-
-	/**
-	 * @deprecated 2022.03
-	 * @see IHandleUserSessions::getLocalUserNickname()
-	 */
-	public function getLoggedInUserNickname(): string
```
Comment (tobiasd) [inline src/App.php:168]: This seems to be used by the calendar, see https://github.com/friendica/friendica/issues/14025#issuecomment-2111605062

**Pull request merged**